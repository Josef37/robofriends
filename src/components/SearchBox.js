import React from "react";

const SearchBox = ({ onSearchChange }) => {
  return (
    <div className="pa2">
      <input
        className="pa3 ba b--green bg-lightest-blue"
        type="search"
        aria-label="Search Robots"
        placeholder="Search Robots"
        onChange={onSearchChange}
      />
    </div>
  );
};

export default SearchBox;
