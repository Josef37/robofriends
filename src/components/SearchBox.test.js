import React from "react";
import { shallow } from "enzyme";
import SearchBox from "./SearchBox";

describe("SearchBox component", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<SearchBox onSearchChange={jest.fn()} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("calls searchChange prop with event when searching", () => {
    const mockSearchChange = jest.fn();
    const wrapper = shallow(<SearchBox onSearchChange={mockSearchChange} />);
    wrapper.find("input").simulate("change", { target: { value: "foo" } });
    expect(mockSearchChange).toBeCalledWith({ target: { value: "foo" } });
  });
});
