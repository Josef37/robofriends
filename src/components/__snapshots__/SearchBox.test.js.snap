// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox component renders correctly 1`] = `
"<div className=\\"pa2\\">
  <input className=\\"pa3 ba b--green bg-lightest-blue\\" type=\\"search\\" aria-label=\\"Search Robots\\" placeholder=\\"Search Robots\\" onChange={[Function: mockConstructor]} />
</div>"
`;
